### ISSUE TITLE
Made.Of() Parameters dont follow Reuse setting

### ISSUE THREAD

**Issue by pamidur** [dadhi/DryIoc#179]
So basically Made.Of() parameters are cached somehow and not refreshed on next call to Transient resolution.

Example:

```c#
class Program
    {
        static void Main(string[] args)
        {
            var container = new Container();
            container.Register<A>();
            container.Register<B>();

            container.Register<ILog>(Made.Of(
                () => LogManager.GetLogger(Arg.Index<Type>(0)),
                request => request.Parent.ImplementationType));

            var b = container.Resolve<B>();

            if (b.Log == b.A.Log)
                Console.WriteLine("Why are they the same instance if ILog is Transient ?");
        }
    }

    public class B
    {
        public ILog Log { get; }
        public A A { get; }

        public B(ILog log, A a)
        {
            Log = log;
            A = a;
        }
    }

    public class A
    {
        public ILog Log { get; }
        public A(ILog log) => Log = log;
    }
```

**Comment by dadhi** [comment]
This is a bug, will look.

**Comment by dadhi** [comment]
I was not able to reproduce it, check the added test here https://github.com/dadhi/DryIoc/commit/ca57802757244e1b243e2b2b659767bbca1c26c5

**Comment by pamidur** [comment]
The thing that isn't updated is Parameter and not the service. I'll make a PR shortly

**Comment by pamidur** [comment]
Ok , I added PR #181 with proper test.
The issue isn't with service instances but rather with its parameters. This parameter was the same for two instances of ILog:

```c#
request => request.Parent.ImplementationType
``` 

It still could be reproduced on 4.0.7 as reported. However I'm happy to inform you that bug isn't there on your latest build (which I think will be 4.10). So this bug should be fixed with next release.

**Comment by dadhi** [comment]
@pamidur , I believe the case should be fixed in v4.1.0-preview-02 already

**Comment by thomasgalliker** [comment]
Huh, great to have found this issue... it looks like it is solved in nuget version 4.1.0-preview-03 (DryIoc.dll nuget). Do you have an idea by when you'll release the stable version of 4.1.x?
